### ISSUE TITLE
Improve validation of needs output object

### ISSUE THREAD

**Issue by Bertg** [rhysd/actionlint#151]
Given the following workflow file:

```yaml
name: Simplified CI example

on: [push]

jobs:
  detect_changes:
    name: "🔎 Detect Changes"
    runs-on: ubuntu-latest
    # outputs will always be strings
    outputs:
      graph: ${{ steps.graph_changes.outputs.change }}
    steps:
      - name: "📫 Checkout with history"
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - id: graph_changes
        name: "🔎 Check for graph changes"
        uses: younited/detect-branch-changes-action@v0.1.3
        with:
          pathspec: "graph/"

  build_graph:
    name: "🧱 Graph"
    needs:
      - detect_changes
    runs-on: ubuntu-latest
    steps:
      - name: "🔬 Example test"
        run: echo OK

  test_graph_dumped_correctly:
    name: "🧪 Graph dumped correctly"
    needs:
      - build_graph
    runs-on: ubuntu-latest
    env:
      GEMS_CACHE_DIR: /tmp/bundle
    steps:
      - name: "🦘 Skipping?"
        if: needs.detect_changes.outputs.graph == 'false'
        run: echo Skipped
      - name: "📫 Checkout"
        if: needs.detect_changes.outputs.graph == 'true'
        uses: actions/checkout@v2
      - name: "🔬 Example test"
        if: needs.detect_changes.outputs.graph == 'true'
        run: echo OK
```

This file passes validation, but has a fairly big bug in it. The job `test_graph_dumped_correctly` uses `needs.detect_changes` but it is not defined in the the `needs` section of that job. 
The bug is subtle, as the jobs will be executed in the expected order (due to the `build_graph` job in between). However the GitHub docs are clear that an explicit `needs` is required. 

> Job outputs are available to all downstream jobs that depend on this job.
https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs#overview

It would be great if this check could be added as well.

**Comment by rhysd** [comment]
This was fixed in [v1.6.17](https://github.com/rhysd/actionlint/releases/tag/v1.6.17).
